### ISSUE TITLE
Demo on images out of datasets

### ISSUE THREAD

**Issue by asimniazi63** [hongsukchoi/Pose2Mesh_RELEASE#3]
I am trying to figure out how to make "input.npy" for any raw image. Can I use posenet (pretrained) to do so?

**Comment by FWC13163296500** [comment]
@asimniazi63 You need to train a two-dimensional joints detection network to identify joints in any image.There are only four types of joints currently supported (human36: 17joints, coco: 19joints, smpl: 24joints, mano: 21joints; for details, refer to demo/run.py).
"--input": 2D joints coordinates
"--res": The resolution of the image where the 2D joints are located (length and width need to be equal)

**Comment by hongsukchoi** [comment]
Hi @asimniazi63 , you need to run independent 2D joint detector to get input for our Pose2Mesh.
For the in-the-wild images, we generally used HRNet (SOTA 2D joint detector) to get coco joints for human body and mano joints for human hand. 
HRNet repo: https://github.com/leoxiaobin/deep-high-resolution-net.pytorch

**Comment by asimniazi63** [comment]
@hongsukchoi Thank for your reply. I was able to get 17 2D joints from the repository you suggested. Now I need to now how can create my own input.npy for human36 or coco 2D joints.
Sample of 2D joints is given below:

```
frame,nose_x,nose_y,left_eye_x,left_eye_y,right_eye_x,right_eye_y,left_ear_x,left_ear_y,right_ear_x,right_ear_y,left_shoulder_x,left_shoulder_y,right_shoulder_x,right_shoulder_y,left_elbow_x,left_elbow_y,right_elbow_x,right_elbow_y,left_wrist_x,left_wrist_y,right_wrist_x,right_wrist_y,left_hip_x,left_hip_y,right_hip_x,right_hip_y,left_knee_x,left_knee_y,right_knee_x,right_knee_y,left_ankle_x,left_ankle_y,right_ankle_x,right_ankle_y
0,209,72,224,62,199,62,245,77,178,77,275,154,143,169,265,261,117,276,168,230,224,250,285,368,301,368,224,577,122,562,199,745,148,740
```

**Comment by hongsukchoi** [comment]
@asimniazi63 
You can check [here](https://github.com/hongsukchoi/Pose2Mesh_RELEASE/blob/master/data/COCO/dataset.py) for the order of joints and to get pelvis and neck joints. I also guided about how to train / test Pose2Mesh [here](https://github.com/hongsukchoi/Pose2Mesh_RELEASE#start).

**Comment by Radhika-Das-Soni** [comment]
> @hongsukchoi Thank for your reply. I was able to get 17 2D joints from the repository you suggested. Now I need to now how can create my own input.npy for human36 or coco 2D joints. Sample of 2D joints is given below:
> 
> ```
> frame,nose_x,nose_y,left_eye_x,left_eye_y,right_eye_x,right_eye_y,left_ear_x,left_ear_y,right_ear_x,right_ear_y,left_shoulder_x,left_shoulder_y,right_shoulder_x,right_shoulder_y,left_elbow_x,left_elbow_y,right_elbow_x,right_elbow_y,left_wrist_x,left_wrist_y,right_wrist_x,right_wrist_y,left_hip_x,left_hip_y,right_hip_x,right_hip_y,left_knee_x,left_knee_y,right_knee_x,right_knee_y,left_ankle_x,left_ankle_y,right_ankle_x,right_ankle_y
> 0,209,72,224,62,199,62,245,77,178,77,275,154,143,169,265,261,117,276,168,230,224,250,285,368,301,368,224,577,122,562,199,745,148,740
> ```

Hi, After we got this, how should we need to use it for training and testing purpose? I found, some of the .yaml files undergoes for test and train, but to make .npy file, we have 2D joints, but not sure how to give it as input to the run.py as you said, that we need to for test and train first to make the joint structure same as given in description but what about .yml file and what about 2D joints which we get from our image?

**Comment by hongsukchoi** [comment]
If you want to train/test on your own dataset, prepare annotation file following [COCO format](https://cocodataset.org/#format-data) and refer to [this](https://github.com/hongsukchoi/Pose2Mesh_RELEASE/blob/master/data/COCO/dataset.py).

Yaml files are not annotatinos files but just for changing hyperparameters or datasets.

**Comment by Radhika-Das-Soni** [comment]
Hi, Thanks for the quick response. I have designed a demo annotation file and compare it with dataset mention in the description of the repository of coco dataset. I would like to confirm that, have you consider one human as one bbox as label? or have you considered, eyes, noes, leg, hand, etc as each bbox and labeled it also as bbox?
